### DISCUSSION TITLE
Create contains filter for numbers

### DISCUSSION THREAD

**Discussion by cristinaAlarcon98** [prisma/prisma#19189] [Q&A] [answered]
I am trying to use the "contains" filter to locate a row in which an input number (as a string) is present within the ID (as an integer) of the row. However, the "contains" filter is intended for use with string data only. Is there a way to achieve the same result with numerical values?

As an illustration, consider the following dataset:

| ID    | Name        | Age | Email              |
|-------|-------------|-----|--------------------|
| 12348 | John Smith  | 42  | john@example.com   |
| 54123 | Jane Doe    | 28  | jane@example.com   |
| 45675 | Bob Johnson | 36  | bob@example.com    |
| 21235 | Alice Green | 31  | alice@example.com  |
| 64567 | Tom Jones   | 45  | tom@example.com    |

Using the input  "123", the expected output would be:

| ID    | Name        | Age | Email             |
|-------|-------------|-----|-------------------|
| 12348 | John Smith  | 42  | john@example.com  |
| 54123 | Jane Doe    | 28  | jane@example.com  |
| 21235 | Alice Green | 31  | alice@example.com |

Thank you in advance!

**Comment by nurul3101** [comment, marked as answer]
Hello @cristinaAlarcon98 👋 

One approach I can think of is to create a separate string representation of the ID column in your database, which you can then use for the `contains` operation, but that wouldn't be the most ideal solution for your case. I think using a raw sql query and casting a string to int and then using a regular expression to match might be a possible solution. I'll let you know in case I can find any better solution.

**Reply by nurul3101** [reply]
@cristinaAlarcon98 It would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 
Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by tophboogie** [comment]
+1  I would also love it if the 'contains' filter could work on numeric database columns. Would this be difficult to implement?

For now I will probably do as @nurul3101 suggested and create a mirrored column that holds the numeric ID as a string, and do a contains on that string column...

**Comment by danbord** [comment]
I'm really surprised I cannot do that with prisma. 
I really need the user to be able to use the contains with a number field, I have way too many.  I also cannot filter after fetching data because I'm using a paging funtionality.
No way I will duplicate all my number fields into strings in the database.

**Comment by kvbc** [comment]
Brotherrrr every single day I'm getting by surprised what functions prisma does NOT have but really should
